backend/functions/sms.js

```diff
@@ -0,0 +1,88 @@
+            if (y = 0, t) op = [op[0] & 2, t.value];
+            switch (op[0]) {
+                case 0: case 1: t = op; break;
+                case 4: _.label++; return { value: op[1], done: false };
+                case 5: _.label++; y = op[1]; op = [0]; continue;
+                case 7: op = _.ops.pop(); _.trys.pop(); continue;
+                default:
+                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
+                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
+                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
+                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
+                    if (t[2]) _.ops.pop();
+                    _.trys.pop(); continue;
+            }
+            op = body.call(thisArg, _);
+        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
+        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
+    }
+};
+//Created Twilio Number, authToken, accountSid; Currently on free trial.
+require('dotenv').config();
+var twilio = require('twilio');
+var accountSid = process.env.TWILIO_ACCOUNT_SID;
+var authToken = process.env.TWILIO_AUTH_TOKEN;
+var client = require('twilio')(accountSid, authToken);
+var twilioNumber = process.env.TWILIO_PHONE_NUMBER;
+function sendSMS(user, message_body) {
+    return __awaiter(this, void 0, void 0, function () {
+        var userPhone, error_1;
+        return __generator(this, function (_a) {
+            switch (_a.label) {
+                case 0:
+                    if (process.env.DATABASE === "staging") {
+                        return [2 /*return*/];
+                    }
+                    userPhone = user.phoneNumber;
+                    if (userPhone === "Dummy number" || userPhone === undefined)
+                        return [2 /*return*/];
+                    _a.label = 1;
+                case 1:
+                    _a.trys.push([1, 3, , 4]);
+                    return [4 /*yield*/, client.messages
+                            .create({
+                            from: twilioNumber,
+                            to: userPhone,
+                            body: message_body,
+                        }).then(function (message) {
+                            (function (message) { return console.log(message.status); });
+                        })];
+                case 2:
+                    _a.sent();
+                    return [3 /*break*/, 4];
+                case 3:
+                    error_1 = _a.sent();
+                    console.log(error_1);
+                    return [3 /*break*/, 4];
+                case 4: return [2 /*return*/];
+            }
```

Comment:
This seems to me to be a very convoluted way of implementing Twilio SMS messaging -- I'd really suggest cleaning it up. You can check out an example of how to do this via this image:
![](https://miro.medium.com/v2/resize:fit:1400/format:webp/1*fZU_XZdt6_B2t4rEC7qTsw.png)

The full article can be found [here](https://javascript.plainenglish.io/how-to-write-a-cloud-function-that-texts-you-dad-jokes-a8e1f2a2dcdc)